soc/arm/renesas_smartbond/da1469x/soc.c

```diff
@@ -26,18 +30,147 @@
 #define FLASH_REGION_SIZE_05M     6
 #define FLASH_REGION_SIZE_025M    7
 
+#define VDD_SLEEP_LEVEL_0V750	0
+#define VDD_CLAMP_LEVEL_0V706	15
+#define V18_LEVEL_1V8		1
+#define V14_LEVEL_1V40		4
+#define V30_LEVEL_3V00		0
+
+#define POWER_CTRL_REG_SET(_field, _val) \
+	CRG_TOP->POWER_CTRL_REG = \
+	(CRG_TOP->POWER_CTRL_REG & ~CRG_TOP_POWER_CTRL_REG_ ## _field ## _Msk) | \
+	((_val) << CRG_TOP_POWER_CTRL_REG_ ## _field ## _Pos)
+
 #if defined(CONFIG_BOOTLOADER_MCUBOOT)
 #define MAGIC 0xaabbccdd
 static uint32_t z_renesas_cache_configured;
 #endif
 
+#if defined(CONFIG_PM) && defined(CONFIG_SOC_DCDC_DA1469X)
+struct dcdc_regs {
+	uint32_t v18;
+	uint32_t v18p;
+	uint32_t vdd;
+	uint32_t v14;
+	uint32_t ctrl1;
+};
+
+static struct dcdc_regs dcdc_state;
+#endif
+
 void sys_arch_reboot(int type)
 {
 	ARG_UNUSED(type);
 
 	NVIC_SystemReset();
 }
 
+void
+z_smartbond_vdd_level_set(uint8_t vdd_level)
+{
+	POWER_CTRL_REG_SET(VDD_LEVEL, vdd_level);
+}
+
+#if defined(CONFIG_SOC_DCDC_DA1469X)
+static void configure_dcdc(void)
+{
+#if defined(CONFIG_SOC_DCDC_DA1469X_1V8)
+	DCDC->DCDC_V18_REG |= DCDC_DCDC_V18_REG_DCDC_V18_ENABLE_HV_Msk;
+	DCDC->DCDC_V18_REG &= ~DCDC_DCDC_V18_REG_DCDC_V18_ENABLE_LV_Msk;
+#else
+	DCDC->DCDC_V18_REG &= ~DCDC_DCDC_V18_REG_DCDC_V18_ENABLE_HV_Msk;
+	DCDC->DCDC_V18_REG &= ~DCDC_DCDC_V18_REG_DCDC_V18_ENABLE_LV_Msk;
+#endif
+
+#if defined(CONFIG_SOC_DCDC_DA1469X_1V8P)
+	DCDC->DCDC_V18P_REG |= DCDC_DCDC_V18P_REG_DCDC_V18P_ENABLE_HV_Msk;
+	DCDC->DCDC_V18P_REG &= ~DCDC_DCDC_V18P_REG_DCDC_V18P_ENABLE_LV_Msk;
+#else
+	DCDC->DCDC_V18P_REG &= ~DCDC_DCDC_V18P_REG_DCDC_V18P_ENABLE_HV_Msk;
+	DCDC->DCDC_V18P_REG &= ~DCDC_DCDC_V18P_REG_DCDC_V18P_ENABLE_LV_Msk;
+#endif
+
+	DCDC->DCDC_VDD_REG |= DCDC_DCDC_VDD_REG_DCDC_VDD_ENABLE_HV_Msk;
+	DCDC->DCDC_VDD_REG |= DCDC_DCDC_VDD_REG_DCDC_VDD_ENABLE_LV_Msk;
+
+	DCDC->DCDC_V14_REG |= DCDC_DCDC_VDD_REG_DCDC_VDD_ENABLE_HV_Msk;
+	DCDC->DCDC_V14_REG |= DCDC_DCDC_VDD_REG_DCDC_VDD_ENABLE_LV_Msk;
+
+#if defined(CONFIG_PM)
+	/* Store DCDC registers to quickly restore DCDC state after wakeup */
+	dcdc_state.v18 = DCDC->DCDC_V18_REG;
+	dcdc_state.v18p = DCDC->DCDC_V18P_REG;
+	dcdc_state.vdd = DCDC->DCDC_VDD_REG;
+	dcdc_state.v14 = DCDC->DCDC_V14_REG;
+	dcdc_state.ctrl1 = DCDC->DCDC_CTRL1_REG;
+#endif
```

Comment:
looks like a regulator, please take a look at regulator.h API

Reply:
I looked at regulator API as requested.
Code you highlighted enables certain power rails according to Kconfig.
Regulator API covers a lot of functionality that is not available in this MCU case.
Code in soc.c uses similar approach as can be found in `nrf52/soc.c` were DCDC is configured as Kconfig dictates.
Please reconsider your change request.

Reply:
Please explain why regulator API is not suitable. There are regulators for e.g. RPi, NXP VREF, etc. nRF code predates regulators, it may need to be revisited.

Reply:
@gmarull, two commits for DCDC configuration removed from this PR. This can be later added as regulators.
Power management functionality implemented in this PR does not require configurability of DCDC.